webpack/assets/javascripts/react_app/redux/actions/statistics.js

```diff
@@ -0,0 +1,19 @@
+import API from '../../API';
+import {
+  STATISTICS_DATA_REQUEST,
+  STATISTICS_DATA_SUCCESS,
+  STATISTICS_DATA_FAILURE
+} from '../consts';
+
+export const getStatisticsData = charts => dispatch => {
+  dispatch({ type: STATISTICS_DATA_REQUEST, payload: charts });
+  charts.map(c => c.url).forEach(url => {
+    API.get(url).then(
+      result => dispatch({ type: STATISTICS_DATA_SUCCESS, payload: result }),
+      (jqXHR, textStatus, errorThrown) => dispatch({
+        type: STATISTICS_DATA_FAILURE,
+        payload: errorThrown
+      })
```

Comment:
error is the line starting with (jqXHR ... 
in then() the first parameter is resolve, the second is reject